ios/RNDeviceInfo/RNDeviceInfo.m

```diff
@@ -76,6 +76,10 @@ - (id)init
     if ((self = [super init])) {
 #if !TARGET_OS_TV
         _lowBatteryThreshold = 0.20;
+
+        self.locationManager = [[CLLocationManager alloc] init];
+        self.locationManager.delegate = self;
```

Comment:
I'm not sure we can do this in the `constantsToExport` area - that code is executed prior to the javascript portion of react-native even starting, meaning there is no ability to do this *optionally* via javascript code - it will run every time. Based on stackoverflow it looks like the locationmanager needs to set up on the  main thread but I think there are other ways to do that, if that is why it's in constantsToExport...

I don't believe app stores and users like it when apps start up location services as a mandatory item 🤔 - I may be misunderstanding but if I am understanding correctly then I think this should be initialized lazily only if ever subscribed, and should be removed both when all subscriptions are closed and on catalyst destroy